hr_employee_transfer/models/hr_department_transfer.py

```diff
@@ -0,0 +1,212 @@
+        states={'draft': [('readonly', False)]}
+    )
+    dst_contract_id = fields.Many2one(
+        'hr.contract',
+        'Contract (TO)',
+        readonly=True
+    )
+    date = fields.Date(
+        'Effective Date',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    note = fields.Text('Reason')
+    state = fields.Selection(
+        [
+            ('draft', 'Draft'),
+            ('confirm', 'Confirmed'),
+            ('pending', 'Pending'),
+            ('done', 'Done'),
+            ('cancel', 'Cancelled'),
+        ],
+        'State',
+        readonly=True,
+        default='draft'
+    )
+
+    _rec_name = 'date'
+
+    _defaults = {
+        'state': 'draft',
+    }
+
+    _track = {
+        'state': _tracked_states,
+    }
+
+    @api.model
+    def _needaction_domain_get(self):
+
+        users_obj = self.env['res.users']
+        domain = []
+        if users_obj.has_group('base.group_hr_manager'):
+            domain = [('state', '=', 'confirm')]
+            return domain
+
+        return False
+
+    @api.multi
+    def unlink(self, cr, uid, ids, context=None):
+        transfers = self.filtered(lambda r: r.state != 'draft')
+        if transfers:
+            raise UserError('Unable to Delete Transfer! \n'
+                            'Transfer has been initiated. Either cancel the '
+                            'transfer or create another transfer to undo it')
+
+        return super(HrDepartmentTransfer, self).unlink()
+
+    @api.onchange('employee_id')
+    @api.one
+    def onchange_employee(self):
+        if self.employee_id:
+            employee = self.employee_id
+            self.src_department_id = employee.department_id and \
+                employee.department_id.id or \
+                False
+            self.src_id = employee.contract_id and \
+                (employee.contract_id.job_id and \
+                employee.contract_id_job_id.id or \
+                False) or False
+            self.src_contract_id = employee.contract_id and \
+                self.employee_id.contract_id.id or False
+
+    @api.model
+    def _check_state(self, contract, effective_date):
+        if contract.date_end and effective_date >= contract.date_end:
+            raise UserError('The contract end date is on or before the '
+                            'effective date of the transfer.')
+        return True
+
+    @api.multi
+    def effective_date_in_future(self):
+        today = fields.Date.today()
+        for xfer in self:
+            if xfer.date <= today:
+                return False
+        return True
+
+    @api.multi
+    def transfer_contract(self, contract, job_id, effective_date):
+        self.ensure_one()
+        # Copy the contract and adjust start/end dates,
+        # job id, etc. accordingly.
+        default = {
+            'job_id': job_id,
+            'date_start': effective_date,
+            'name': False,
```

Comment:
Hi @tarzan0820 this code will produce error unless we depend this module to hr_contract_reference. What do you think we should do:

1. assign default value for name (e.g. /)
2. depend to hr_contract_reference so that module will automatically create sequence for new contract